### ISSUE TITLE
Specifying different color for bash prompt

### ISSUE THREAD

**Issue by dieterdemeyer** [jonmosco/kube-ps1#31]
Hello,

I'm trying to specify a different color for the Kubernetes context according to the documentation for bash but this doesn't seem to work.
Would you be able to assist me with this ?
I basically want to modify the context color from red to purple.

Thanks.

Regards,
Dieter.

**Comment by jonmosco** [comment]
Can you paste your configuration please?

**Comment by jonmosco** [comment]
Bash or zsh?

**Comment by dieterdemeyer** [comment]
I'm using bash

**Comment by jonmosco** [comment]
This is actually an enhancement that I have to work on for bash.  I am in the process of unifying the color handling so it will be the same for bash and zsh.  I will label this as an enhancement for now.  
This change should be in the next day or two as well.

Thanks again for making me aware of this issue and using kube-ps1.

**Comment by jonmosco** [comment]
Try this new commit out and let me know if it does what you expect.

**Comment by dieterdemeyer** [comment]
I've just tried your new code but it seems to be behaving differently..
Without setting any colors explicitly by ENV variables, I'm getting the following prompt
```
(k8s|docker-for-desktop:default) » ~
```
instead of previously
```
(⎈ |docker-for-desktop:default) » ~
```

Before:
![image](https://user-images.githubusercontent.com/523286/35769069-c4277826-0905-11e8-9742-a231cfb01470.png)

After:
![image](https://user-images.githubusercontent.com/523286/35769071-c810791a-0905-11e8-8bcf-77ba191e65e2.png)

**Comment by jonmosco** [comment]
can you send me the output of this:  

export | grep KUBE_PS1

**Comment by dieterdemeyer** [comment]
![1 bash 2018-02-03 17-54-43](https://user-images.githubusercontent.com/523286/35769452-5a5e1386-090b-11e8-9fcb-ef6d8b135656.png)

**Comment by jonmosco** [comment]
Oh, sorry, was looking at this on my phone.  What is your $LANG set to?

echo $LANG

**Comment by dieterdemeyer** [comment]
```
(k8s|docker-for-desktop:default) » ~ echo $LANG
en_US.UTF-8
```

**Comment by jonmosco** [comment]
Check out the README as well, you can now specify color names as mentioned, and depending on your setup, the color code can be used as well.

**Comment by dieterdemeyer** [comment]
I've tried with one of the color names but the color remained the same..
But I'll try with the codes instead.

**Comment by jonmosco** [comment]
Can you paste your config for kube-ps1 here?  Bash version, etc as well.

**Comment by jonmosco** [comment]
I suspect this is another bash < 4 issue.  No echo -e, or printing with \u support, so I added hex codes for bash < 4.  Try it out and let me know.  If the colors are still and issue, the codes might need changed to.

**Comment by dieterdemeyer** [comment]
I actually have no specific config for kube-ps1, I only source your sh script and call the function in my prompt.

My bash version is as follows:
```
(k8s|docker-for-desktop:default) » ~ bash --version
GNU bash, version 4.4.12(1)-release (x86_64-apple-darwin16.3.0)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by jonmosco** [comment]
ok thanks for the info.  Did the newest commit from today help?

**Comment by dieterdemeyer** [comment]
I replaced my kube-ps1.sh script with the new version but the result was unfortunately the same.

**Comment by jonmosco** [comment]
Ok, well, we can try a couple more things.  What does `echo $KUBE_PS1_SHELL` return?

The only reason `k8s` will appear is if it fails the shell check condition and doesn't print the symbols.

**Comment by dieterdemeyer** [comment]
```
(k8s|docker-for-desktop:default) » ~ echo $KUBE_PS1_SHELL
bash
```

It's strange because in the previous version of the code, the symbol would indeed print.

**Comment by jonmosco** [comment]
That is strange.  The only real difference is the usage of the actual symbol codes as opposed to putting the symbol itself in the script (this was done to support older versions of Bash).  

Can you try to unset the `KUBE_PS1_SYMBOL`, then start a new shell and source the script again ?  Perhaps the variable is not being overwritten correctly?

**Comment by dieterdemeyer** [comment]
Something like this you mean ?

![1 bash 2018-02-04 19-05-50](https://user-images.githubusercontent.com/523286/35780620-7efc4932-09de-11e8-8a30-ac1286c64f0e.png)

**Comment by jonmosco** [comment]
I am having trouble reproducing this on MacOS with iTerm2:

<img width="780" alt="screen shot 2018-02-04 at 1 20 35 pm" src="https://user-images.githubusercontent.com/1970496/35780768-5a36272e-09ae-11e8-9050-12c1b96ed965.png">

**Comment by dieterdemeyer** [comment]
I've tried again with the script before I updated it with your changes regarding bash improvements.
The previous script seems to be working as expected.
But using the latest version seems to result in my errors..  Really strange...

**Comment by jonmosco** [comment]
can you paste `echo $PS1`

**Comment by dieterdemeyer** [comment]
```
(k8s|docker-for-desktop:default) » ~ echo $PS1
$(kube_ps1) \[\033[0;32m\]\302\273\[\033[0m\] \[\033[0;34m\]\w\[\033[0m\]\[\033[0m\]
```

**Comment by jonmosco** [comment]
This is a strange one indeed:

<img width="1003" alt="screen shot 2018-02-04 at 4 27 11 pm" src="https://user-images.githubusercontent.com/1970496/35782604-5b540b98-09c8-11e8-8471-d29afef89732.png">

**Comment by jonmosco** [comment]
Can you try to start bash without your configuration, and source kube-ps1?

```
bash --norc
source /path/to/kube-ps1.sh
PS1='$(kube_ps1) \[\033[0;32m\]\302\273\[\033[0m\] \[\033[0;34m\]\w\[\033[0m\]\[\033[0m\] '
```

**Comment by jonmosco** [comment]
Also, another question.  Can you show me the output of echo trying to print the character:

```
$ echo -e $'\u2388'
```

**Comment by dieterdemeyer** [comment]
![1 bash 2018-02-05 19-39-15](https://user-images.githubusercontent.com/523286/35822350-4c8e099e-0aac-11e8-856a-723328bf8b20.png)

**Comment by dieterdemeyer** [comment]
This is weird because the first line where the cluster context is in fact red, it's when sourcing the kube-ps1.sh without the latest changes to bash colors...

**Comment by jonmosco** [comment]
What is the CTX color before and after:

`echo $KUBE_PS1_CTX_COLOR`

**Comment by jonmosco** [comment]
Also, is there anything left over from your original question because you wanted it to be set to purple?

"I basically want to modify the context color from red to purple."

**Comment by dieterdemeyer** [comment]
I wanted to easily customise the colours, yes.
But that's not working at the moment... I'll try to reproduce this in a clean environment because I have a lot of stuff in my bash environment, this to see if something else is causing this...

**Comment by jonmosco** [comment]
Ok, just making sure something isn't left over from it previously is all.

What did you get for the output of `echo $KUBE_PS1_CTX_COLOR` ?

**Comment by dieterdemeyer** [comment]
The output for that command is `red`.

**Comment by jonmosco** [comment]
Ok as it should be for the default, so its strange the purple color.

**Comment by jonmosco** [comment]
Any luck?

**Comment by dieterdemeyer** [comment]
As a matter of fact, yes.
I started with a clean bash, and it seems to work fine with the latest version of the script.
So something must be messing it up with my current configuration.
I also noticed that the ENV variables need to be set before sourcing the script, which is obvious in retrospect...

I'll debug some more to find exactly what's causing the problem but the issue can be closed as far as I'm concerned.

Thanks for the help in debugging.
